TiledLib/TmxMisc.cs

```diff
@@ -79,25 +80,32 @@ static void ReadTile(this XmlReader reader, Dictionary<int, Dictionary<string, s
             if (!tileProperties.TryGetValue(id, out properties) || properties == null)
                 properties = tileProperties[id] = new Dictionary<string, string>();
 
-            reader.ReadStartElement("tile");
-            while (reader.IsStartElement())
-                switch (reader.Name)
-                {
-                    case "properties":
-                        reader.ReadProperties(properties);
-                        break;
-                    case "animation":
-                        tileAnimations[id] = reader.ReadAnimation();
-                        break;
-                    default:
-                        reader.Skip(); //TODO: Add logging.
-                        break;
-                }
-
-            if (reader.Name == "tile")
-                reader.ReadEndElement();
+            if (reader.IsEmptyElement)
```

Comment:
Some of the sample files I had have xml like:

    <tile />

XmlReader doesn't generate EndElement's for empty elements, so the code was going crazy.